### PR TITLE
Upgrade Kotlin to 1.9.20 and Ktlint to 1.0.1

### PR DESCRIPTION
**Pull request by jdisanti**
This PR upgrades Kotlin to 1.9.20, and Ktlint to 1.0.1.

I had to abandon the pretty-format-kotlin pre-commit hook since it was ignoring our `.editorconfig`, which made it impossible to suppress some of the newer lints that would be difficult and undesirable to fix. Now all Ktlint runs go through gradle, so there isn't any way for inconsistencies to creep in.

I recommend reviewing this PR commit by commit as the commit that runs Ktlint against the entire repo has a very large diff. When reviewing the Ktlint formatting changes, be sure to hide whitespace changes to reduce how much needs to be examined.

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

**Comment by rcoh** [comment]
the codegen changes appear to be whitespace only although the whitespace ignoring link is broken

**Review by rcoh — APPROVED** [review]

**Review by 82marbag — APPROVED** [review]

**Comment by jdisanti** [comment]
I unintentionally merged this into jdisanti-upgrade-nightly rather into main. Going to post a new PR to merge this to the correct branch once jdisanti-upgrade-nightly is merged to main.

**Pull request merged**
